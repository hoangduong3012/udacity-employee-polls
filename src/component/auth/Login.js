import { connect } from "react-redux";
import { useState } from "react";
import { handleUserLogin } from "../../actions/authUser";
import { useNavigate } from "react-router-dom";
import { useLocation } from "react-router-dom";
import NavigationBar from "../common/NavigationBar";
import { Footer } from "../common/Footer";
import Button from "react-bootstrap/Button";
import Card from 'react-bootstrap/Card';
import { Container } from "react-bootstrap";

const Login = (props) => {
  const navigate = useNavigate();
  const [username, setUsername] = useState("hoangbui");
  const [password, setPassword] = useState("password123");
  const { state } = useLocation();

  const handleSubmit = (e) => {
    props.dispatch(handleUserLogin(username, password));
    navigate(state?.path || "/");
  };
  return (
    <div>
      <div className="d-flex justify-content-center login-wrapper bg-grey text-dark p-5 text-center text-sm-start">
        <div className="Container">
          <div className="d-sm-flex justify-content-between">
            <div className="Auth-form-containe">
              <form className="Auth-form" onSubmit={handleSubmit}>
                <div className="Auth-form-content">
                  <h3 className="Auth-form-title">Sign In</h3>
                  <div className="form-group mt-3">
                    <label>Username</label>
                    <input
                      type="text"
                      data-testid="username"
                      value={username}
                      onChange={(e) => setUsername(e.target.value)}
                      className="form-control mt-1"
                      placeholder="Enter username"
                      autoComplete="on"
                    />
                  </div>
                  <div className="form-group mt-3">
                    <label>Password</label>
                    <input
                      type="password"
                      data-testid="password"
                      name="email"
                      value={password}
                      onChange={(e) => setPassword(e.target.value)}
                      className="form-control mt-1"
                      placeholder="Enter password"
                      autoComplete="on"
                    />
                  </div>
                  <div className="d-grid gap-2 mt-3">
                    <Button 
                      data-testid="submitButton"
                      type="submit"
                      className="btn btn-primary bg-dark"
                    >
                      Submit
                    </Button>
                  </div>
                  <p className="forgot-password text-right mt-2">
                    Forgot <a href="#">password?</a>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
};

const mapStateToProps = ({ dispatch, authedUser }) => ({
  dispatch,
  authedUser,
});
export default connect(mapStateToProps)(Login);
