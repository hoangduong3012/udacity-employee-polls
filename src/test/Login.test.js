import Login from "../component/auth/Login";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../store";
import { BrowserRouter as Router } from "react-router-dom";

describe("App", () => {
  it("will match snapshot", () => {
    var component = render(
      <Provider store={store}>
        <Router>
        <Login />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});

it('should check that submit input is active', () => {
  const { getByPlaceholderText } = render( <Provider store={store}>
    <Router>
    <Login />
    </Router>
  </Provider>)
  const inputElement = screen.getByPlaceholderText("Enter username");
 fireEvent.change(inputElement, { target: {value: "type a username"}})
 expect(inputElement.value).toBe("type a username")
})
