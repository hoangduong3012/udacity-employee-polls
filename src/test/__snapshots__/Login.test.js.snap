// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="d-flex justify-content-center login-wrapper bg-grey text-dark p-5 text-center text-sm-start"
        >
          <div
            class="Container"
          >
            <div
              class="d-sm-flex justify-content-between"
            >
              <div
                class="Auth-form-containe"
              >
                <form
                  class="Auth-form"
                >
                  <div
                    class="Auth-form-content"
                  >
                    <h3
                      class="Auth-form-title"
                    >
                      Sign In
                    </h3>
                    <div
                      class="form-group mt-3"
                    >
                      <label>
                        Username
                      </label>
                      <input
                        autocomplete="on"
                        class="form-control mt-1"
                        data-testid="username"
                        placeholder="Enter username"
                        type="text"
                        value="hoangbui"
                      />
                    </div>
                    <div
                      class="form-group mt-3"
                    >
                      <label>
                        Password
                      </label>
                      <input
                        autocomplete="on"
                        class="form-control mt-1"
                        data-testid="password"
                        name="email"
                        placeholder="Enter password"
                        type="password"
                        value="password123"
                      />
                    </div>
                    <div
                      class="d-grid gap-2 mt-3"
                    >
                      <button
                        class="btn btn-primary bg-dark btn btn-primary"
                        data-testid="submitButton"
                        type="submit"
                      >
                        Submit
                      </button>
                    </div>
                    <p
                      class="forgot-password text-right mt-2"
                    >
                      Forgot 
                      <a
                        href="#"
                      >
                        password?
                      </a>
                    </p>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
        <div>
          <footer
            class="bgg"
          >
            <div
              class="footer-contents container"
            >
              <div>
                <a
                  class="childs nav-link"
                  href="/"
                >
                  Home
                </a>
                <a
                  class="childs nav-link"
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
                <a
                  class="childs nav-link"
                  href="/addPoll"
                >
                  New
                </a>
                <a
                  class="childs nav-link"
                  href="/FAQs"
                >
                  FAQs
                </a>
                <a
                  class="childs nav-link"
                  href="/addPoll"
                >
                  About
                </a>
              </div>
              <hr />
              <div>
                © 2024 Hoang Bui Company, Inc
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="d-flex justify-content-center login-wrapper bg-grey text-dark p-5 text-center text-sm-start"
      >
        <div
          class="Container"
        >
          <div
            class="d-sm-flex justify-content-between"
          >
            <div
              class="Auth-form-containe"
            >
              <form
                class="Auth-form"
              >
                <div
                  class="Auth-form-content"
                >
                  <h3
                    class="Auth-form-title"
                  >
                    Sign In
                  </h3>
                  <div
                    class="form-group mt-3"
                  >
                    <label>
                      Username
                    </label>
                    <input
                      autocomplete="on"
                      class="form-control mt-1"
                      data-testid="username"
                      placeholder="Enter username"
                      type="text"
                      value="hoangbui"
                    />
                  </div>
                  <div
                    class="form-group mt-3"
                  >
                    <label>
                      Password
                    </label>
                    <input
                      autocomplete="on"
                      class="form-control mt-1"
                      data-testid="password"
                      name="email"
                      placeholder="Enter password"
                      type="password"
                      value="password123"
                    />
                  </div>
                  <div
                    class="d-grid gap-2 mt-3"
                  >
                    <button
                      class="btn btn-primary bg-dark btn btn-primary"
                      data-testid="submitButton"
                      type="submit"
                    >
                      Submit
                    </button>
                  </div>
                  <p
                    class="forgot-password text-right mt-2"
                  >
                    Forgot 
                    <a
                      href="#"
                    >
                      password?
                    </a>
                  </p>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div>
        <footer
          class="bgg"
        >
          <div
            class="footer-contents container"
          >
            <div>
              <a
                class="childs nav-link"
                href="/"
              >
                Home
              </a>
              <a
                class="childs nav-link"
                href="/leaderboard"
              >
                Leaderboard
              </a>
              <a
                class="childs nav-link"
                href="/addPoll"
              >
                New
              </a>
              <a
                class="childs nav-link"
                href="/FAQs"
              >
                FAQs
              </a>
              <a
                class="childs nav-link"
                href="/addPoll"
              >
                About
              </a>
            </div>
            <hr />
            <div>
              © 2024 Hoang Bui Company, Inc
            </div>
          </div>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
